hoomd/pytest/test_simulation.py

```diff
@@ -329,3 +329,50 @@ def test_seed(simulation_factory, lattice_snapshot_factory):
 
     sim.seed = 20
     assert sim.seed == 20
+
+
+def test_operations_setting(simulation_factory, lattice_snapshot_factory):
+    sim = simulation_factory()
+    sim.create_state_from_snapshot(lattice_snapshot_factory())
+    old_operations = sim.operations
+    operations = hoomd.Operations()
+    # Add some operations to test the setting
+    operations += hoomd.update.BoxResize(
+        hoomd.Box.cube(10),
+        hoomd.Box.cube(20),
+        hoomd.variant.Ramp(0, 1, 0, 100),
+        40)
+    operations += hoomd.write.GSD("foo.gsd", 10)
+    operations += hoomd.write.Table(
+        10, logger=hoomd.logging.Logger(['scalar']))
+    operations.tuners.clear()
+    # Check before and after setting
+    assert sim.operations is not operations
+    assert operations._simulation is None
+    assert old_operations._simulation is sim
+    sim.operations = operations
+    assert sim.operations is operations
+    assert operations._simulation is sim
+    assert old_operations._simulation is None
+
+    sim.run(0)
+    # Check setting after scheduling
+    new_operations = hoomd.Operations()
+    new_operations += hoomd.update.BoxResize(
+        hoomd.Box.cube(300),
+        hoomd.Box.cube(20),
+        hoomd.variant.Ramp(0, 1, 0, 100),
+        80)
+    new_operations += hoomd.write.GSD("bar.gsd", 20)
+    new_operations += hoomd.write.Table(
+        20, logger=hoomd.logging.Logger(['scalar']))
+
+    assert sim.operations is not new_operations
+    assert sim.operations._simulation is sim
+    assert new_operations._simulation is None
+    sim.operations = new_operations
+    assert sim.operations is new_operations
+    assert new_operations._simulation is sim
+    assert new_operations._scheduled
+    assert not operations._scheduled
+    assert operations._simulation is None
```

Comment:
Many of these checks are done earlier in this test as well, perhaps some of these checks could be put in their own helper method? It could be called `_assert_before_after_reassign_operations`, or something like that.